src/test/java/org/opensrp/web/rest/ClientFormResourceTest.java

```diff
@@ -316,55 +315,53 @@ public void testSearchForFormByFormVersionShouldReturnNextJsonValidatorVersion()
         assertEquals("0.0.3", jsonNode.get("clientFormMetadata").get("version").textValue());
         assertTrue(jsonNode.get("clientFormMetadata").get("isJsonValidator").booleanValue());
     }
-
+    
     @Test
     public void testAddClientFormWhenGivenJSON() throws Exception {
         String formIdentifier = "opd/reg.json";
-        String formVersion = "0.1.1";
         String formName = "REGISTRATION FORM";
 
         MockMultipartFile file = new MockMultipartFile("form", "path/to/opd/reg.json",
                 "application/json", TestFileContent.JSON_FORM_FILE.getBytes());
 
+        when(manifestService.getAllManifest(anyInt())).thenReturn(getManifestList());
         when(clientFormService.addClientForm(any(ClientForm.class), any(ClientFormMetadata.class))).thenReturn(mock(ClientFormService.CompleteClientForm.class));
 
         mockMvc.perform(
                 fileUpload(BASE_URL)
                         .file(file)
                         .param("form_identifier", formIdentifier)
-                        .param("form_version", formVersion)
                         .param("form_name", formName))
                 .andExpect(status().isCreated())
                 .andReturn();
 
         ArgumentCaptor<ClientForm> clientFormArgumentCaptor = ArgumentCaptor.forClass(ClientForm.class);
         ArgumentCaptor<ClientFormMetadata> clientFormMetadataArgumentCaptor = ArgumentCaptor.forClass(ClientFormMetadata.class);
         verify(clientFormService).addClientForm(clientFormArgumentCaptor.capture(), clientFormMetadataArgumentCaptor.capture());
+        verify(manifestService).getAllManifest(anyInt());
 
         assertEquals(TestFileContent.JSON_FORM_FILE, clientFormArgumentCaptor.getValue().getJson().toString());
         ClientFormMetadata clientFormMetadata = clientFormMetadataArgumentCaptor.getValue();
         assertEquals(formIdentifier, clientFormMetadata.getIdentifier());
-        assertEquals(formVersion, clientFormMetadata.getVersion());
+        assertEquals("0.0.2", clientFormMetadata.getVersion());
         assertEquals(formName, clientFormMetadata.getLabel());
         assertNull(clientFormMetadata.getModule());
     }
 
-    @Test
-    public void testAddClientFormWhenGivenJSONValidatorFile() throws Exception {
+    @Test public void testAddClientFormWhenGivenJSONValidatorFile() throws Exception {
         String formIdentifier = "opd/reg.json";
-        String formVersion = "0.1.1";
```

Comment:
This were removed since my understanding was, we aren't making the form version optional but removing the ability for a user/dev to actually set one.  In that case we don't need to pass in a form version.